src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -816,9 +816,21 @@ public bool RemoveTransientTransactions(uint256 depositId = null)
                 // Remove transient transactions not seen in a block yet.
                 bool walletUpdated = false;
 
-                foreach ((Transaction transaction, IWithdrawal withdrawal) in this.FindWithdrawalTransactions(depositId))
+                if (depositId == null)
+                {
+                    foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions.ToList())
+                    {
+                        if (transactionData.BlockHeight == null)
+                        {
+                            this.Wallet.MultiSigAddress.Transactions.Remove(transactionData);
```

Comment:
Shouldn't this be calling RemoveTransaction instead?

Reply:
We specifically want to remove all TransactionData that was not seen in a block without adding any conditions and relying on minimum state.

Reply:
But don't we need to update spent transactions too? By calling RemoveTransactions?

Reply:
If the `BlockHeight == null` it won't be spent.

Reply:
Okay that's a fair assumption, but I still would have thought that RemoveTransactions would be wiser because the wallet could right itself if it ever did manage to get into a situation where it accidentally spent an unconfirmed

Reply:
I did not want to make the `RemoveTransaction` method public (with accompanying locking) just for this or have to to find a depositId to pass to `RemoveTransientTransactions` .

Reply:
Since `RemoveTransaction` removes a spent `Transaction` and not a `TransactionData` it requires the `outpointLookup` which I would rather not use at this point if not required.

Reply:
>If the BlockHeight == null it won't be spent.  

Are you sure about that?

Reply:
I tend to agree with @codingupastorm why not use the existing RemoveTransaction method otherwise you just create more places where logic is applied.